azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -93,33 +103,44 @@ protected void createFunctionApp() throws Exception {
         } else {
             final ExistingAppServicePlanWithGroup planWithGroup = functionApp.withExistingAppServicePlan(plan);
             withCreate = isResourceGroupExist(resGrp) ?
-                    planWithGroup.withExistingResourceGroup(resGrp) :
-                    planWithGroup.withNewResourceGroup(resGrp);
+                planWithGroup.withExistingResourceGroup(resGrp) :
+                planWithGroup.withNewResourceGroup(resGrp);
         }
         configureAppSettings(withCreate::withAppSettings, getAppSettings());
-        withCreate.create();
+        withCreate.withJavaVersion(DEFAULT_JAVA_VERSION).withWebContainer(null).create();
 
         info(String.format(FUNCTION_APP_CREATED, getAppName()));
     }
 
     protected void updateFunctionApp(final FunctionApp app) {
         info(FUNCTION_APP_UPDATE);
-
         // Work around of https://github.com/Azure/azure-sdk-for-java/issues/1755
         app.inner().withTags(null);
-
         final Update update = app.update();
+        checkFunctionHostJavaVersion(app, update); // Check Java Version of Server
         configureAppSettings(update::withAppSettings, getAppSettings());
         update.apply();
-
         info(FUNCTION_APP_UPDATE_DONE + getAppName());
     }
 
+    protected void checkFunctionHostJavaVersion(final FunctionApp app, final Update update) {
+        final JavaVersion serverJavaVersion = app.javaVersion();
+        if (serverJavaVersion.toString().matches(VALID_JAVA_VERSION_PATTERN)) {
+            info(String.format(FUNCTION_HOST_JAVA_VERSION, serverJavaVersion));
+        } else if (serverJavaVersion.equals(JavaVersion.OFF)) {
+            info(FUNCTION_HOST_JAVA_VERSION_OFF);
+            update.withJavaVersion(DEFAULT_JAVA_VERSION);
+        } else {
+            warning(FUNCTION_HOST_JAVA_VERSION_INCORRECT);
+            update.withJavaVersion(DEFAULT_JAVA_VERSION);
+        }
```

Comment:
merge these with one brace and a conditional expr:
 else if (serverJavaVersion.equals(JavaVersion.OFF)) {
            info(FUNCTION_HOST_JAVA_VERSION_OFF);
            update.withJavaVersion(DEFAULT_JAVA_VERSION);
        } else {
            warning(FUNCTION_HOST_JAVA_VERSION_INCORRECT);
            update.withJavaVersion(DEFAULT_JAVA_VERSION);
        }